File: src/main/java/com/lcaohoanq/nocket/domain/reaction/PostReactionDTO.java
package com.lcaohoanq.nocket.domain.reaction;

import com.lcaohoanq.nocket.enums.EReaction;
import jakarta.validation.constraints.NotNull;
import java.util.UUID;

public record PostReactionDTO(
    @NotNull(message = "Post Id is required") UUID postId,
    EReaction reaction
) { }

File: server/src/main/kotlin/com/lcaohoanq/nocket/domain/reaction/PostReactionDTO.kt
package com.lcaohoanq.nocket.domain.reaction

import com.lcaohoanq.nocket.enums.EReaction
import jakarta.validation.constraints.NotNull
import java.util.*

data class PostReactionDTO(
    val postId: @NotNull(message = "Post Id is required") UUID,
    val reaction: EReaction
)
